src/database/queries/AddressQueries.js

```diff
@@ -128,13 +166,60 @@ class AddressQueries {
 				LIMIT
 					$3
 				OFFSET
-					$4;
+					$4
+			`,
+			values: [
+				hexToBytea(address),
+				hexToBytea(contract),
+				limit,
+				offset
+			]
+		}
+	}
+
+	static getTokensForContractAmount(
```

Comment:
I think we can optimize this. It's fine for a starting point, but I get the feeling that we can more accurately get the number of tokens needed for an address and contract without needing to pull the entire record set simply to count the number of rows.

Reply:
Resolved with simpler query.